app/models/discussion.rb

```diff
@@ -48,8 +47,16 @@ class Discussion < ActiveRecord::Base
 
   has_many :events, -> { includes :user }, as: :eventable, dependent: :destroy
 
-  has_many :items, -> { includes(eventable: :user).where(kind: THREAD_ITEM_KINDS).order('created_at ASC') }, class_name: 'Event'
-  has_many :salient_items, -> { includes(eventable: :user).where(kind: SALIENT_ITEM_KINDS).order('created_at ASC') }, class_name: 'Event'
+  has_many :items, -> { includes(eventable: :user).where(kind: Event::THREAD_ITEM_KINDS).order('created_at ASC') }, class_name: 'Event'
+  has_many :salient_items, -> { includes(eventable: :user).where(kind: Event::SALIENT_ITEM_KINDS).order('created_at ASC') }, class_name: 'Event'
+
+  define_counter_cache(:items_count)         { |discussion| discussion.items.count }
```

Comment:
This is the simplest implementation, but I'm not certain it's the most performant, since it means we'll be loading up all the items every time an event is created / destroyed for the discussion... I might like to do something like

```
if item_is_created? then discussion.items_count + 1 else discussion.items_count - 1 end
```
But not quite sure how to know whether the callback's being called on create or destroy from the block yet.